lib/compiler/src/beam_validator.erl

```diff
@@ -243,9 +243,15 @@ build_function_table([{function,Name,Arity,Entry,Code0}|Fs], Acc) ->
                      end, Code0),
     case Code of
         [{label,Entry} | Is] ->
-            Info = #{ name => Name,
-                      arity => Arity,
-                      parameter_info => find_parameter_info(Is, #{}) },
+            Info0 = #{ name => Name,
+                       arity => Arity,
+                       parameter_info => find_parameter_info(Is, #{}) },
+            Info = case Is of
+                       [{jump,_}|_] ->
+                           Info0#{always_fails => true};
+                       _ ->
+                           Info0
+                   end,
```

Comment:
Nitpick: I prefer having these properties explicit, maybe break this out into another function like `find_parameter_info/2`?

```suggestion
                       parameter_info => find_parameter_info(Is, #{}),
                       always_fails => find_always_fails(Is) },
```